src/ng/rootScope.js

```diff
@@ -892,6 +894,94 @@ function $RootScopeProvider() {
+       *   transcluded scope will also be suspended, even if the scope from which the transcluded
+       *   scope inherits is not suspended.
+       * * Multiple directives trying to manage the suspended status of a scope can confuse each other:
+       *    * A call to `$suspend` an already suspended scope is a no-op.
+       *    * A call to `$resume` a non-suspended scope is a no-op.
+       *    * If two directives suspend a scope, then one of them resumes the scope, the scope will no
+       *      longer be suspended. This could result in the other directive believing a scope to be
+       *      suspended when it is not.
+       * * If a parent scope is suspended then all its descendants will be excluded from future digests
+       *   whether or not they have been suspended themselves. Note that this also applies to isolate
+       *   child scopes.
+       * * Calling `$resume()` on a scope that has a suspended ancestor will not cause the scope to be
+       *   included in future digests until all its ancestors have been resumed.
+       * * Resolved promises, e.g. from explicit `$q` deferreds and `$http` calls, trigger `$apply`
+       *   against the `$rootScope` and so will still trigger a global digest even if the promise was
+       *   initialized by a component that lives on a suspended scope.
+       */
+      $suspend: function() {
+        this.$$suspended = true;
+      },
+
+      /**
+       * @ngdoc method
+       * @name $rootScope.Scope#$isSuspended
+       * @kind function
+       *
+       * @description
+       * Call this method to determine if this scope or one of its ancestors have been suspended.
+       *
+       * Be aware that a scope may not be included in digests if it has a suspended ancestor,
+       * even if `$isSuspended()` returns false.
+       *
+       * To determine if this scope will be excluded in digests then you must check all its ancestors:
+       *
+       * ```
+       * function isExcludedFromDigest(scope) {
+       *   while(scope) {
+       *     if (scope.$isSuspended()) return true;
+       *     scope = scope.$parent;
+       *   }
+       *   return false;
+       * ```
+       *
+       * @returns true if the current scope has been suspended.
+       */
+      $isSuspended: function() {
+        return this.$$suspended;
+      },
+
+      /**
+       * @ngdoc method
+       * @name $rootScope.Scope#$resume
+       * @kind function
+       *
+       * @description
+       * Resume watchers of this scope subtree in case it was suspended.
+       *
+       * It is recommended to digest on this scope after it is resumed to catch any modifications
+       * that might have happened while it was suspended.
+       *
+       * See {@link $rootScope.Scope#$suspend} for information about the dangers of using this approach.
+       */
+      $resume: function() {
+        this.$$suspended = false;
+      },
```

Comment:
IMO, `$resume` should also set `dirty = true` and `lastDirtyWatch = null` (in case already during a `$digest`).

Reply:
If `$resume` triggered an async apply then we wouldn't need to worry about this, no?

Reply:
Correct (I think :grin:).

Reply:
It appears that calling resume from a watch handler is fine, because the fact that the watch handler was called means that the digest is already marked as dirty. So a new digest has already been triggered.

Reply:
And we don't seem to need to worry about the short circuiting of `lastDirtyWatch`:
* if the resumed scope is "after" the last dirty watch, then it will have been digested on that turn of the digest cycle.
* if the resumed scope is "before" the last dirty watch, then it will get digested before we hit the short circuit.

Reply:
So we don't need to run any `applyAsyc` etc.

Reply:
Interesting! 👍